lib/segment/src/index/hnsw_index/graph_layers_builder.rs

```diff
@@ -836,4 +919,69 @@ mod tests {
         graph_layers_builder.links_map(0, 0, |link| result.push(link));
         assert_eq!(&result, &vec![1, 2, 3, 4, 5, 6]);
     }
+
+    #[test]
+    fn test_optimized_heuristic() {
```

Comment:
Are we keeping both the optimized + unoptimized versions available? Though it's nice that we can compare them for making sure they behave the same

Reply:
This change is very sensitive and I have to be sure that graph didn't change